Classes/EventStore/Storage/Doctrine/DoctrineEventStorage.php

```diff
@@ -79,125 +74,72 @@ public function load(string $streamName)
             ->addOrderBy('version', 'ASC')
             ->setParameter('stream', $streamName, \PDO::PARAM_STR);
 
-        $data = $this->unserializeEvents($query);
-
-        if ($data === []) {
-            return null;
-        }
-
-        $cacheKey = md5($streamName . '.' . $version);
-        $this->runtimeCache[$cacheKey] = new EventStreamData($data, $version);
-
-        return $this->runtimeCache[$cacheKey];
+        $streamIterator = new DoctrineStreamIterator($query->execute());
+        return new EventStream($streamIterator, $this->getStreamVersion($streamName));
     }
 
     /**
      * @param string $streamName
-     * @param array $data
-     * @param int $commitVersion
-     * @param \Closure $callback
-     * @return int
+     * @param WritableToStreamInterface $events
+     * @param int $expectedVersion
+     * @return void
      * @throws ConcurrencyException|\Exception
      */
-    public function commit(string $streamName, array $data, int $commitVersion, \Closure $callback = null)
+    public function commit(string $streamName, WritableToStreamInterface $events, int $expectedVersion = ExpectedVersion::ANY)
```

Comment:
> It just somehow doesn't fit to the rest of our naming scheme.

I agree.. Also I don't know if we really need to have an interface to cover one vs many events. It could also just be:
```php
/**
 * @param string $streamName
 * @param WritableEvent[] $events
 * @param int $expectedVersion
 **/
public function commit(string $streamName, array $events, int $expectedVersion = ExpectedVersion::ANY)
```

(or another name, but just an array of x)